packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -146,26 +146,17 @@ const ElementHook: Hooks<VElement> = {
     },
 };
 
-function isUpgradableComponent(
-    ctor: CustomElementConstructor | UpgradableCustomElementConstructor
-): ctor is UpgradableCustomElementConstructor {
-    return 'upgrade' in ctor;
-}
-
 const CustomElementHook: Hooks<VCustomElement> = {
     create: (vnode) => {
         const {
             sel,
             owner: { renderer },
         } = vnode;
         const UpgradableConstructor = getUpgradableConstructor(sel, renderer);
-        if (isUpgradableComponent(UpgradableConstructor)) {
+        const elm = new UpgradableConstructor((elm: HTMLElement) => {
             // the custom element from the registry is expecting an upgrade callback
-            UpgradableConstructor.upgrade = (elm: HTMLElement) => {
```

Comment:
instead of setting it into the upgrade setter from super, we just pass the callback when `new`ing it